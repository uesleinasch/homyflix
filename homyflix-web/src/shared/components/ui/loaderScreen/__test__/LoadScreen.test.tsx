import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import LoaderScreen from '..';
import { MantineProvider } from '@mantine/core';

describe('LoaderScreen', () => {
    beforeAll(() => {
        renderWithProvider();
    });

  const renderWithProvider = (props = {}) => {
    return render(
      <MantineProvider>
        <LoaderScreen {...props} />
      </MantineProvider>
    );
  };

  it('Should render correctly with default props', () => {
    const { container } = renderWithProvider();
    expect(container).toMatchSnapshot();
  });

  it('Should not render when isLoading is false', () => {
    const { container } = renderWithProvider({ isLoading: false });
    
    expect(container).not.toContain('LoaderScreen');
  });

  it('Should render correctly when isLoading is true', () => {
    renderWithProvider({ isLoading: true });
    
    const progressbar = screen.getByRole('progressbar');
    expect(progressbar).toBeInTheDocument();
    expect(progressbar).toHaveAttribute('aria-label', 'Carregando aplicação');
    expect(progressbar).toHaveAttribute('aria-busy', 'true');
    
    expect(screen.getByText('Carregando...')).toBeInTheDocument();
  });

  it('Should render with custom loading text', () => {
    const customText = 'Aguarde um momento...';
    renderWithProvider({ isLoading: true, loadingText: customText });
    
    expect(screen.getByText(customText)).toBeInTheDocument();
    
    expect(screen.queryByText('Carregando...')).not.toBeInTheDocument();
  });

  it('Should have correct CSS classes and structure', () => {
    renderWithProvider({ isLoading: true });
    
    const progressbar = screen.getByRole('progressbar');
    
    expect(progressbar).toHaveClass('overlay');
    
    const loader = progressbar.querySelector('.mantine-Loader-root');
    expect(loader).toBeInTheDocument();
  });
});