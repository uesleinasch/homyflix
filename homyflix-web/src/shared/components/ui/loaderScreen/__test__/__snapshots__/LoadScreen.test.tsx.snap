// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderScreen Should render correctly with default props 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    aria-busy="true"
    aria-label="Carregando aplicação"
    class="overlay"
    role="progressbar"
  >
    <div
      class="container"
    >
      <span
        class="m_4e3f22d7 m_5ae2e3c mantine-Loader-root"
        data-size="xl"
        style="--loader-size: var(--loader-size-xl); --loader-color: var(--mantine-color-orange-filled);"
      >
        <span
          class="m_870c4af"
        />
        <span
          class="m_870c4af"
        />
        <span
          class="m_870c4af"
        />
      </span>
      <span
        class="text"
      >
        Carregando...
      </span>
    </div>
  </div>
</div>
`;
